### PR TITLE
Solving white space + improving code +PEP8

### PR DESCRIPTION
**Pull request by alexandrejaguar**
Hi everyone,
please check if this is you wanted.
Thank you very much.

### CODE REVIEW

**Comment by stefanv** [comment]
Thanks for working on this, Alex!  Can we change the axes names to (ax0, ax1, ax2) please?

Also, please join us on https://gitter.im/scikit-image/scikit-image

**Comment by alexandrejaguar** [comment]
Hi Stéfan,
I put it that way because this example have two subplots. I will change it back.
How do I do that? Pulling the file again?
Thank you very much!

File: doc/examples/plot_line_hough_transform.py [hunk 1]
```diff
@@ -64,70 +64,71 @@
...
 idx = np.arange(25, 75)
 image[idx[::-1], idx] = 255
 image[idx, idx] = 255
 
+# Classic straight-line Hough transform.
 h, theta, d = hough_line(image)
 
-fig, (ax1, ax2, ax3) = plt.subplots(1, 3, figsize=(8,4))
+# Generating figure 1.
+fig, (ax1, ax2, ax3) = plt.subplots(1, 3, figsize=(12, 6))
```
Comment (emmanuelle) [inline doc/examples/plot_line_hough_transform.py]: this is where @stefanv suggested to replace with ``(ax0, ax1, ax2)`` (since Python is counting from 0). Other occurrences below.

File: doc/examples/plot_line_hough_transform.py [hunk 2]
```diff
@@ -64,70 +64,71 @@
...
+           d[-1], d[0]], cmap=cm.gray, aspect=1/1.5)
 ax2.set_title('Hough transform')
 ax2.set_xlabel('Angles (degrees)')
 ax2.set_ylabel('Distance (pixels)')
 ax2.axis('image')
 
-ax3.imshow(image, cmap=plt.cm.gray)
-rows, cols = image.shape
+ax3.imshow(image, cmap=cm.gray)
+row1, col1 = np.shape(image)
```
Comment (emmanuelle) [inline doc/examples/plot_line_hough_transform.py]: I tend to prefer ``image.shape`` (the objected-oriented way of calling ``shape``) to ``np.shape(image)``. What was your reason for changing it?

Also, why did you change ``rows, cols``?

File: doc/examples/plot_line_hough_transform.py [hunk 3]
```diff
@@ -64,70 +64,71 @@
...
 
-ax2.imshow(edges, cmap=plt.cm.gray)
+ax2.imshow(edges, cmap=cm.gray)
 ax2.set_title('Canny edges')
 ax2.set_axis_off()
 ax2.set_adjustable('box-forced')
 
-ax3.imshow(edges * 0)
-
+ax3.imshow(edges*0)
```
Comment (emmanuelle) [inline doc/examples/plot_line_hough_transform.py]: ``(edges * 0)`` looks slightly more PEP8-compliant to me.

**Comment by emmanuelle** [comment]
I left a few comments that should be easy to address, and then it will be good to go! Thanks!

File: doc/examples/plot_line_hough_transform.py (hunk 1 above)
Comment (alexandrejaguar) [inline doc/examples/plot_line_hough_transform.py]: OK, I'll change that.

File: doc/examples/plot_line_hough_transform.py (hunk 2 above)
Comment (alexandrejaguar) [inline doc/examples/plot_line_hough_transform.py]: I think np.shape(image) is more clear, but you have a point. I'll change it back.

`rows, cols` assumes two values in this example, each one used for a image set. In order to adjust the image size, I stablished two variable sets: `row1, col1` and `row2, col2`.

File: doc/examples/plot_line_hough_transform.py (hunk 3 above)
Comment (alexandrejaguar) [inline doc/examples/plot_line_hough_transform.py]: Sure thing, I'll change it too.

**Comment by alexandrejaguar** [comment]
Hi @emmanuelle,
I solved the issues on `plot_line_hough_transform.py`. Hope it is ok now. I'll work on the another files, ok?
Thank you very much!

File: doc/examples/plot_line_hough_transform.py [hunk 4]
```diff
@@ -64,70 +64,71 @@
...
     y0 = (dist - 0 * np.cos(angle)) / np.sin(angle)
-    y1 = (dist - cols * np.cos(angle)) / np.sin(angle)
-    ax3.plot((0, cols), (y0, y1), '-r')
-ax3.axis((0, cols, rows, 0))
-ax3.set_title('Detected lines')
-ax3.set_axis_off()
-
-# Line finding, using the Probabilistic Hough Transform
+    y1 = (dist - col1 * np.cos(angle)) / np.sin(angle)
+    ax3.plot((0, col1), (y0, y1), '-r')
```
Comment (emmanuelle) [inline doc/examples/plot_line_hough_transform.py]: ``ax2.plot`` (``ax3`` does not exist any more).

File: doc/examples/plot_line_hough_transform.py [hunk 5]
```diff
@@ -64,70 +64,71 @@
...
-ax2.set_title('Canny edges')
-ax2.set_axis_off()
-ax2.set_adjustable('box-forced')
-
-ax3.imshow(edges * 0)
-
+ax2.imshow(edges * 0)
 for line in lines:
     p0, p1 = line
     ax3.plot((p0[0], p1[0]), (p0[1], p1[1]))
```
Comment (emmanuelle) [inline doc/examples/plot_line_hough_transform.py]: Same here.

**Comment by emmanuelle** [comment]
Hi @alexandrejaguar thanks for the changes. If you can just remove the remaining ``ax3``, I think ``plot_line_hough_transform.py`` is fine as it is. You can apply the same changes to the other files, thanks!

File: doc/examples/plot_line_hough_transform.py (hunk 3 above)
Comment (alexandrejaguar) [inline doc/examples/plot_line_hough_transform.py]: Hi Emmanuelle,
I found the answer on this one. According to PEP-8, spaces on operations can be used to separate operations with different precedence, such as `sum(a*b + c*d)`. I believe we don't need the spaces when we have only one operation happening.

**Comment by alexandrejaguar** [comment]
Hi @emmanuelle, sorry about that silly mistake. I changed it and also `plot_ssim.py`. Could you check that? Thank you very much, sorry about wasting your time.

**Comment by emmanuelle** [comment]
No worries @alexandrejaguar that's the whole purpose of code review :-). 

In the meantime, after you started your PR, PR #1823 has been merged into master: this PR moved gallery examples to subdirectories in order to have more structure in the gallery. Could you please rebase on master (following instructions on http://scikit-image.org/docs/stable/contribute.html#divergence-between-upstream-master-and-your-feature-branch) to remove existing conflicts? Thanks!

Otherwise, your changes look good to me, but to be sure it's better to have Travis CI results!

**Comment by alexandrejaguar** [comment]
@emmanuelle there was another editing file on my repo. I'll try to rebase now and work with Travis, ok?

**Comment by emmanuelle** [comment]
@alexandrejaguar I'm not sure to understand what you mean by "another editing file on my repo". If you rebase, conflicts will be solved, and Travis will run (build the packages and the doc, run the tests etc.). Travis does it automatically, so the only thing that you have to do is to see whether Travis builds pass (green button) or if they fail, where the error comes from.

**Comment by alexandrejaguar** [comment]
@emmanuelle I pushed another file on my local repository, `plot_local_otsu.py`. But there are errors yet...
I followed the steps on the instructions:

```bash
jaguar@pepperoni:~/Dropbox/git-repos/scikit-image$ git checkout master
Already on 'master'
Your branch is up-to-date with 'origin/master'.
jaguar@pepperoni:~/Dropbox/git-repos/scikit-image$ git pull
Already up-to-date.
jaguar@pepperoni:~/Dropbox/git-repos/scikit-image$ git rebase
First, rewinding head to replay your work on top of it...
Fast-forwarded master to refs/remotes/origin/master.
jaguar@pepperoni:~/Dropbox/git-repos/scikit-image$ git push
Username for 'https://github.com': alexandrejaguar
Password for 'https://alexandrejaguar@github.com': 
Everything up-to-date
```

**Comment by alexandrejaguar** [comment]
@emmanuelle I did it all wrong. I'll restart my copy as you teach on http://scikit-image.org/docs/stable/contribute.html. Sorry for all trouble.

**Comment by alexandrejaguar** [comment]
Now I got it :) Awesome!

**Comment by stefanv** [comment]
Before merging this PR, please make sure the changes got compressed into one commit.

**Comment by alexandrejaguar** [comment]
Hi @stefanv, how could I do that? Thanks!

**Comment by blink1073** [comment]
Hi @alexandrejaguar, the incantation is as follows:

```bash
$ git rebase -i
# Select squash or 's' for all but the first commit and then write a summary commit message
$ git push origin -f
```

**Comment by emmanuelle** [comment]
@blink1073 should we add these instructions to the "contribute" section of the user guide?

**Comment by blink1073** [comment]
Yep, but I also messed it up a bit.

```bash
$ git fetch upstream master
$ git rebase -i upstream/master
# Select squash or 's' for all but the first commit and then write a summary commit message
$ git push origin -f
```

**Comment by alexandrejaguar** [comment]
@blink1073 thank you very much, I'll do this asap.

**Comment by emmanuelle** [comment]
Hum, not quite yet, there should be only one commit remaining in the stack. Did you select 'p' (pick) for the first commit and 's' (squash) for all other commits?

**Comment by alexandrejaguar** [comment]
Hi @emmanuelle,
I'm using another PC, so I made all the ritual again. Now I commited as you and @blink1073 said.
Thank you very much.

**Comment by emmanuelle** [comment]
@alexandrejaguar congratulations and thanks! Merging.

File: doc/examples/segmentation/plot_local_otsu.py [hunk 6]
```diff
@@ -10,23 +10,21 @@
 
 The example compares the local threshold with the global threshold.
 
-.. note: local is much slower than global thresholding
+.. Note: local is much slower than global thresholding
```
Comment (stefanv) [inline doc/examples/segmentation/plot_local_otsu.py]: I think this should have been ``.. note:: local is much slower...``

Comment (alexandrejaguar) [inline doc/examples/segmentation/plot_local_otsu.py]: @stefanv I'll fix that. Please let me know if there is anything wrong.

**Comment by alexandrejaguar** [comment]
@emmanuelle @stefanv thanks! I've learned a lot on the process. I'll read more about skimage style and info on contributions. After that I'll continue to fix the other examples. Is that ok? Thanks again!

**Comment by stefanv** [comment]
@alexandrejaguar That will be very much OK :)  Thank you!
